monai/data/image_reader.py

```diff
@@ -690,14 +694,18 @@ class WSIReader(ImageReader):
     def __init__(self, backend: str = "OpenSlide", level: int = 0):
         super().__init__()
         self.backend = backend.lower()
-        if self.backend == "openslide":
-            self.wsi_reader, *_ = optional_import("openslide", name="OpenSlide")
-        elif self.backend == "cucim":
-            self.wsi_reader, *_ = optional_import("cucim", name="CuImage")
-        elif self.backend == "tifffile":
-            self.wsi_reader, *_ = optional_import("tifffile", name="TiffFile")
-        else:
-            raise ValueError('`backend` should be "cuCIM", "OpenSlide", or "TiffFile')
+
+        @require_pkg(pkg_name=self.backend)
```

Comment:
Why hiding this API, we lose the flexibility of inheriting it easily

Reply:
Hi @wyli ,

I think it's not supported in python to set `pkg_name=self.backend` directly if not hiding the function in `init()`:
```py
def __init__(self, backend: str = "OpenSlide", level: int = 0):
    super().__init__()
    self.backend = backend.lower()
    self.wsi_reader = self._set_reader(self.backend)
    self.level = level

@require_pkg(pkg_name=self.backend)
def _set_reader(self, backend: str):
    if backend == "openslide":
        return OpenSlide
    if backend == "cucim":
        return CuImage
    if backend == "tifffile":
        return TiffFile
    raise ValueError("`backend` should be 'cuCIM', 'OpenSlide' or 'TiffFile'.")
```

Maybe I missed some python feature?

Thanks.

Reply:
I think you can use the decorator in the constructor as a normal function "require_pkg(backend)(self._set_reader)"

Reply:
updated in the latest commit.

Thanks.